packages/yaml/src/yaml-document.ts

```diff
@@ -38,10 +38,16 @@ export class YamlDocument {
      * @memberof YamlDocument
      */
     private constructor(serializedYaml: string) {
-        this.document = yaml.parseDocument(serializedYaml);
-        if (this.document.errors?.length > 0) {
+        this.documents = yaml.parseAllDocuments(serializedYaml);
+        if (this.documents.length === 0) {
+            // this is done for backward compatibility when only single document yaml was supported.
+            // yaml.parseDocument('') creates a new document
+            // yaml.parseAll('') does not create a new document
+            this.documents.push(yaml.parseDocument(serializedYaml));
+        }
+        if (this.documents[0].errors?.length > 0) {
```

Comment:
Consider errors in the other documents as well?

Reply:
added it with https://github.com/SAP/open-ux-tools/pull/906/commits/ce25fc781ab51ad569d1dfd679865a49d4d199c3